DOLDatabase/WhereClause.cs

```diff
@@ -0,0 +1,265 @@
+﻿/*
+ * DAWN OF LIGHT - The first free open source DAoC server emulator
+ * 
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ *
+ */
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DOL.Database
+{
+    public abstract class WhereClause
+    {
+        private static string alphabet = "abcdefghijklmnopqrstuvwxyz";
+
+        private string GetPlaceHolder(uint id)
+        {
+            var prefix = "@";
+            uint radix = (uint)alphabet.Length;
+            var result = "";
+            do
+            {
+                var letter = alphabet[(int)(id % radix)];
+                result = letter + result;
+                id /= radix;
+            } while (id > 0);
+            return prefix + result;
+        }
+
+        public virtual string ParameterizedText
```

Comment:
It should not be here, the SQL syntax can be different between 2 dbms so the IR to SQL should be done in the SQL implementation.

I think it's the whole purpose of using WhereClause instead of sql string directly if it can't do that, all theses changes are way less interesting.

Reply:
The purpose of this is just to simplify the old parameterized queries. If you someday decide to use a completely different (Non-SQL) DBMS you can still refactor it, but otherwise YAGNI applies.

Reply:
`Type` is a reserved keyword and it's used at least for the Ban table so it's needed.

Reply:
What is the scenario you're testing there?

Reply:
I didn't tested it, I'm just saying that this PR is adding complexity (and possible bugs with it) for nearly nothing if it can't manage simple scenario like having a sql keyword as a column name.

If I follow the YAGNI principle, this whole PR is against it: it add complexity for doing less. When you write the sql where clause directly, you can do some subqueries, use sql functions, etc. And you don't need to learn a new way to query the database.

Reply:
As I said above the purpose of the original change was to simplify them and that includes easing the transition to parameterized queries. The process of writing them was error prone and repetitive with no IDE help. In addition the WhereClause queries are also easier to read.

Reply:
Ok, I see. Sorry for losing your time about that!